src/libstore/local-store.cc

```diff
@@ -221,7 +221,16 @@ LocalStore::LocalStore(ref<const Config> config)
        schema upgrade is in progress. */
     if (!config->readOnly) {
         Path globalLockPath = dbDir + "/big-lock";
-        globalLock = openLockFile(globalLockPath.c_str(), true);
+        try {
+            globalLock = openLockFile(globalLockPath.c_str(), true);
+        } catch (SysError & e) {
+            if (e.errNo == EACCES || e.errNo == EPERM) {
+                e.addTrace({},
+                    "This command may have been run as non-root in a single-user Nix installation,\n"
```

Comment:
I'm confused by the tense ("may have been") here. Is the *current* command being run as non-root, or was a previous command run as non-root?

Reply:
we could probably also check check "isNonRootuser()" to make a more concrete option.

Reply:
I guess we could say "you may want to start a Nix daemon", but the user doesn't really connect to the Nix daemon themselves so I don't think "you may want to connect to a Nix daemon instead" would be very helpful. Did you have something in mind?